cache4k/src/commonTest/kotlin/io/github/reactivecircus/cache4k/CacheBuilderTest.kt

```diff
@@ -14,29 +14,29 @@ class CacheBuilderTest {
     @Test
     fun expireAfterWrite_zeroDuration() {
         val exception = assertFailsWith<IllegalArgumentException> {
-            Cache.Builder()
+            Cache.Builder<Any, Any>()
                 .expireAfterWrite(0.nanoseconds)
-                .build<Any, Any>()
```

Comment:
I noticed you added generics to the `Builder` itself. Are there any benefits?

Reply:
also if we end up with this API change, please update the `README.md`.
A new section to showcase the event listener API would be great but feel free to add it as a follow up.

Reply:
> I noticed you added generics to the `Builder` itself. Are there any benefits?
I moved the types to `Builder` because `eventListener(listener: CacheEventListener<K, V>)` needs the same types as the `build` method.

Reply:
A disadvantage of this change is that the following no longer infers types for `K` and `V`:

```kotlin
val cache: Cache<A, B> = Cache.Builder().build()
```

Perhaps `@BuilderInference` could help here?

Reply:
Your case can be solved with a kotlin-style builder:
```kotlin
fun <K : Any, V : Any> buildCache(block: Cache.Builder<K, V>.() -> Unit): Cache<K, V> {
    return Cache.Builder<K, V>().apply(block).build()
}

val cache: Cache<Long, String> = buildCache {
    expireAfterAccess(5.minutes)
    eventListener { event ->
        println("onEvent $event")
    }
}
```